src/Test.FunctionalTests.Sarif/Multitool/ValidateCommandTests.cs

```diff
@@ -389,16 +340,26 @@ protected override string ConstructTestOutputFromInputResource(string inputResou
 
             string actualLogFileContents = File.ReadAllText(actualLogFilePath);
             SarifLog actualLog = JsonConvert.DeserializeObject<SarifLog>(actualLogFileContents);
-
-            // First, we'll strip any validation results that don't originate with the rule under test
-            var newResults = new List<Result>();
-
-            foreach (Result result in actualLog.Runs[0].Results)
+            Run run = actualLog.Runs[0];
+
+            // First, we'll strip any validation results that don't originate with the rule under test.
+            // But leave the results that originate from JSchema! Also, be careful because output files
+            // from "valid" test cases don't have any results.
+            run.Results = run.Results
+                ?.Where(r => IsRelevant(r.RuleId, ruleUnderTest))
+                ?.ToList();
+
+            // Next, remove any rule metadata for those rules. The output files from "valid" test
+            // cases don't have any rules.
+            run.Tool.Driver.Rules = run.Tool.Driver.Rules
+                ?.Where(r => IsRelevant(r.Id, ruleUnderTest))
+                ?.ToList();
+
+            // Since there's only one rule in the metadata, the ruleIndex for all remaining results
+            // must be 0.
+            foreach (Result result in run.Results)
             {
-                if (result.RuleId == ruleUnderTest)
-                {
-                    newResults.Add(result);
```

Comment:
>newResults [](start = 20, length = 10)

There already _was_ logic to remove results from other rules -- but it was never hooked up (the `newResults` list wasn't assigned back to the log file). In this PR, we hook it up, and we make additional necessary adjustments to the results and the rules.